packages/amplify-environment-parameters/src/environment-parameter-manager.ts

```diff
@@ -37,10 +37,10 @@ export const getEnvParamManager = (envName: string = stateManager.getLocalEnvInf
 /**
  * Execute the save method of all currently initialized IEnvironmentParameterManager instances
  */
-export const saveAll = async (): Promise<void> => {
+export const saveAll = async (serviceUploadHandler?: ServiceUploadHandler): Promise<void> => {
   for (const envParamManager of Object.values(envParamManagerMap)) {
     // save methods must be executed in sequence to avoid race conditions writing to the tpi file
-    await envParamManager.save();
+    await envParamManager.save(serviceUploadHandler);
```

Comment:
Why pass the serviceUploadHandler into the save function instead of the constructor of an implementation of the `IEnvironmentParameterManager` interface?